test/addons/esm/test-import.js

```diff
@@ -0,0 +1,8 @@
+// Flags: --experimental-addon-modules
+
+'use strict';
+const common = require('../../common');
+
+import('./test-esm.mjs')
```

Comment:
Since this is re-using the CJS module cache, can you add some tests for testing import + require & require + import of the same addon to make sure re-loading from different paths work correctly?